### ISSUE TITLE
getDocumentInfo performance

### ISSUE THREAD

**Issue by marekhrabe** [adobe-photoshop/generator-core#257]
(tldr is at the end :)))

Everytime I request data by `getDocumentInfo`, Photoshop freezes for some time (like half a second on my Mac for medium sized PSD). Is there a way around this? Is there some way to call it without blocking scrolling and other functionality? 

It is my current problem in CSS Hat where I use `getDocumentInfo` as a way to get detailed data about document to generate code for selected layers. I need to update this document info everytime I get `imageChanged` event (and `metaDataOnly` is undefined), so I'm working with latest data.

There are some things (probably visibilities and bounds) that I can merge without problem to cached document from `imageChanged` event data and I will probably do that as soon as possible and release an update for Hat users, but there are some moments when reloading data is necesary - like when user changes font and I get only `text: true` so I know that text changed, but I just don't know how. 

**tldr:** Can we make getDocumentInfo non-blocking? Or is there a way to incrementally update "all" document metadata (including `getFullTextStyles` etc.) without asking for whole document? Maybe a `getLayerInfo` API that would be probably much faster than whole document? (just a guess, I don't know what is causing the slowdown internally) Or do you have another idea?

**Comment by marekhrabe** [comment]
I will close this in favour of #299 which will probably solve these issues.
